### ISSUE TITLE
Incompatibility with Marcel 1.0.3

### ISSUE THREAD

**Issue by toby-brilliant** [carrierwaveuploader/carrierwave#2729]
Marcel released recently, for the first time in years. As the project mentions [here](https://github.com/rails/marcel/issues/98#issuecomment-1972266795), it seems CarrierWave was relying on a private constant which changed.

This line: https://github.com/carrierwaveuploader/carrierwave/blob/ed8799191824a4d2762eb1028c01220102699377/lib/carrierwave/downloader/remote_file.rb#L36

does not work with the newly released Marcel 1.0.3 as `Marcel::TYPES` has been changed. This yields stack traces that look something like:

```
     NameError:
       uninitialized constant Marcel::TYPES
     # ./app/models/attachment.rb:44:in `block in set_remote_url'
     # ./app/models/attachment.rb:40:in `set_remote_url
 ```
any time `set_remote_url` is used.

For anyone stuck on this, you can add Marcel directly to your Gemfile and pin it to the last version (e.g. `gem "marcel", "=1.0.2"`)

**Comment by fwolfst** [comment]
Note that jeremy gives instructions about how to fix this in https://github.com/rails/marcel/issues/98#issuecomment-1973839866 .

**Comment by bvogel** [comment]
fixed by https://github.com/carrierwaveuploader/carrierwave/pull/2728 

luckily I found/suggested the same fix that @jeremy suggested.

**Comment by Muscul** [comment]
Please upgrade gem version and apply this fix in rubgems.

**Comment by mshibuya** [comment]
Released 3.0.6.
https://rubygems.org/gems/carrierwave/versions/3.0.6
